common/src/main/java/com/wynntils/wynn/model/quests/QuestManager.java

```diff
@@ -0,0 +1,190 @@
+ */
+package com.wynntils.wynn.model.quests;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.managers.CoreManager;
+import com.wynntils.core.webapi.WebManager;
+import com.wynntils.core.webapi.request.Request;
+import com.wynntils.core.webapi.request.RequestBuilder;
+import com.wynntils.core.webapi.request.RequestHandler;
+import com.wynntils.mc.objects.Location;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.Utils;
+import com.wynntils.utils.WebUtils;
+import com.wynntils.wynn.event.QuestBookReloadedEvent;
+import com.wynntils.wynn.event.TrackedQuestUpdateEvent;
+import com.wynntils.wynn.event.WorldStateEvent;
+import com.wynntils.wynn.model.WorldStateManager;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import net.minecraft.world.phys.Vec3;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class QuestManager extends CoreManager {
+    public static final QuestScoreboardHandler SCOREBOARD_HANDLER = new QuestScoreboardHandler();
+    private static final QuestContainerQueries CONTAINER_QUERIES = new QuestContainerQueries();
+    private static final DialogueHistoryQueries DIALOGUE_HISTORY_QUERIES = new DialogueHistoryQueries();
+
+    private static List<QuestInfo> quests = List.of();
+    private static List<QuestInfo> miniQuests = List.of();
+    private static List<List<String>> dialogueHistory = List.of();
+    private static QuestInfo currentQuest = null;
+
+    public static void init() {}
+
+    @SubscribeEvent(priority = EventPriority.HIGH)
+    public static void onWorldStateChanged(WorldStateEvent e) {
+        quests = List.of();
+        dialogueHistory = List.of();
+    }
+
+    @SubscribeEvent(priority = EventPriority.LOWEST)
+    public static void onWorldStateChangedLast(WorldStateEvent e) {
+        // Rescan quest book in background as the last thing we do
+        // when entering a world
+        if (e.getNewState() == WorldStateManager.State.WORLD) {
+            CONTAINER_QUERIES.queryQuestBook();
+        }
+    }
+
+    public static void rescanQuestBook(boolean includeQuests, boolean includeMiniQuests) {
+        WynntilsMod.info("Requesting rescan of Quest Book");
+        if (includeQuests) {
+            CONTAINER_QUERIES.queryQuestBook();
+        }
+        if (includeMiniQuests) {
+            CONTAINER_QUERIES.queryMiniQuests();
+        }
+    }
+
+    public static void rescanDialogueHistory() {
+        DIALOGUE_HISTORY_QUERIES.scanDialogueHistory();
+    }
+
+    public static List<QuestInfo> getQuests() {
+        return quests;
+    }
+
+    public static List<QuestInfo> getMiniQuests() {
+        return miniQuests;
+    }
+
+    public static List<QuestInfo> getQuestsSorted(QuestSortOrder sortOrder) {
+        return sortQuestInfoList(sortOrder, quests);
+    }
+
+    public static List<QuestInfo> getMiniQuestsSorted(QuestSortOrder sortOrder) {
```

Comment:
Rename to have `Sorted` before `MiniQuests`/`Quests`?

Reply:
Actually, this should be the main way to get quests; the "raw" methods are (currently) not even used. I removed those and skipped the "sorted" part. If we ever need raw access again, let's name it something with "raw" instead.